src/app/legacy/components/Promo/image.jsx

```diff
@@ -47,29 +42,27 @@ const createSizes = () => {
 
 const Image = props => {
   const { children, src, ...rest } = props;
-
   const isWebPSupported = WEBP_ORIGIN_CODES.some(originCode =>
     src.includes(originCode),
   );
+  const primarySrcSet = createSrcSet(src, isWebPSupported ? '.webp' : '');
+  const primaryMediaType = `image/${isWebPSupported ? 'webp' : 'jpeg'}`;
+  const fallbackSrcSet = isWebPSupported ? createSrcSet(src) : undefined;
+  const fallbackMediaType = isWebPSupported ? 'image/jpeg' : undefined;
```

Comment:
Not so much for this ticket, but I wonder should this work be done in the new Image component, as it seems like all images will need to do this calculations.

Reply:
@aaronmoran I think that makes the component less generic and maybe doing too much, which is a problem with so many of our components. There are also some exceptions like with SVGs and the topic image badges that are just simple non-responsive images with just a src. It makes sene to have 1 helper function to create the srcsets though and passing the result of that function into the component is fine I think

Reply:
Yea that sounds fair. I remember using the utility in `src/app/lib/utilities/srcSet/index.js` to create the srcsets when working on the WebP stuff, so if that can be used as the central place for generating the srcsets that would be good. Its used a lot in tandem with `buildIChefURL` in `src/app/lib/utilities/ichefURL/index.js` as well.